templates/injector-network-policy.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.openshift }}
```

Comment:
This can apply to vanilla Kubernetes indeed, however I didn't want to impact with this PR the way how Vault is currently deployed on vanilla k8s.
Regarding the the second suggestion, yes you are right. I will update this accordingly